src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -644,6 +655,133 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table2, indicies_sub);
+
+            return dest;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void AdvSimdDecode(ref byte* srcBytes, ref byte* destBytes, byte* srcEnd, int sourceLength, int destLength, byte* srcStart, byte* destStart)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian);
+
+            // Complete lookup table - similar to that used in the SS3 decode.
+            Vector128<byte> dec_lut0 = Vector128.Create(255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255);
+            Vector128<byte> dec_lut1 = Vector128.Create(255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255);
+            Vector128<byte> dec_lut2 = Vector128.Create(255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,  62, 255, 255, 255,  63);
+            Vector128<byte> dec_lut3 = Vector128.Create( 52,  53,  54,  55,  56,  57,  58,  59,  60,  61, 255, 255, 255, 255, 255, 255);
+            Vector128<byte> dec_lut4 = Vector128.Create(255,   0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,  14);
+            Vector128<byte> dec_lut5 = Vector128.Create( 15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25, 255, 255, 255, 255, 255);
+            Vector128<byte> dec_lut6 = Vector128.Create(255,  26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  40);
+            Vector128<byte> dec_lut7 = Vector128.Create( 41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51, 255, 255, 255, 255, 255);
+
+            // Interleave pattern for the ST3.
+            Vector128<byte> st3_interleave_index0 = Vector128.Create((byte) 0, 16, 32,  1, 17, 33,  2, 18, 34,  3, 19, 35,  4, 20, 36,  5);
+            Vector128<byte> st3_interleave_index1 = Vector128.Create((byte)21, 37,  6, 22, 38,  7, 23, 39,  8, 24, 40,  9, 25, 41, 10, 26);
+            Vector128<byte> st3_interleave_index2 = Vector128.Create((byte)42, 11, 27, 43, 12, 28, 44, 13, 29, 45, 14, 30, 46, 15, 31, 47);
+
+            // Some constants.
+            Vector128<byte> vzero = Vector128.Create((byte)0);
+            Vector128<byte> v255 = Vector128.Create((byte)255U);
+            Vector128<byte> v16 = Vector128.Create((byte)16U);
+
+            byte* src =  srcBytes;
+            byte* dest = destBytes;
+
+            do
+            {
+                // Load 64 bytes of data and deinterleave the result.
+                // This is equivalent to a NEON LD4 instruction.
+                Vector128<byte> str0 = Vector128.LoadUnsafe(ref *src);
+                Vector128<byte> str1 = Vector128.LoadUnsafe(ref *src, 16);
+                Vector128<byte> str2 = Vector128.LoadUnsafe(ref *src, 32);
+                Vector128<byte> str3 = Vector128.LoadUnsafe(ref *src, 48);
```

Comment:
I suspect you can use `AdvSimd.Arm64.LoadPairVector128` (and even nontemporal if it makes sense) here - jit is not smart enough yet to do it by itself

Reply:
Doing this worked, but didn't give any improvement